test/index.js

```diff
@@ -253,78 +251,121 @@ describe("RedisCache", function () {
       done();
     });
   });
 
-  it("should set the value in redis with ttl in nearest seconds", function (done) {
+  it("should set the value in redis with ttl in nearest seconds", function(done) {
     var target = new RedisCache();
-    target.set("key", "value", 1500).then(function () {
+    target.set("key", "value", 1500).then(function() {
       target.client.cache.key.ttl.should.equal(2);
       done();
     });
   });
 
-  it("should not set the value in redis if ttl is less than 1", function (done) {
+  it("should not set the value in redis if ttl is less than 1", function(done) {
     var target = new RedisCache();
-    target.set("key", "value", 0).then(function () {
+    target.set("key", "value", 0).then(function() {
       assert(target.client.cache.key === undefined);
       done();
     });
   });
 
-  it("should set the value in redis permanently if without ttl", function (done) {
+  it("should set the value in redis permanently if without ttl", function(done) {
     var target = new RedisCache();
-    target.set("key", "value").then(function () {
+    target.set("key", "value").then(function() {
       assert(target.client.cache.key.ttl === null);
       done();
     });
   });
 
-  it("should set the value in redis with default ttl if without ttl", function (done) {
+  it("should set the value in redis permanently if maxAge is empty string", function(done) {
+    var target = new RedisCache({ maxAge: "" });
+    target.set("key", "value");
+
+    assert(target.client.cache.key.ttl === null);
+    done();
+  });
+
+  it("should set the value in redis with a ttl if given in constructor options", function(done) {
     var target = new RedisCache({ maxAge: 1000 });
-    target.set("key", "value").then(function () {
+    target.set("key", "value").then(function() {
       target.client.cache.key.ttl.should.equal(1);
       done();
     });
   });
 
-  it("should set the value in redis as json", function (done) {
+  it("should set the value in redis with a specifically given ttl that overrides global default values", function(done) {
+    var target = new RedisCache({ maxAge: 1000 });
+    target.set("key", "value", 2000).then(function() {
+      target.client.cache.key.ttl.should.equal(2);
+      done();
+    });
+  });
+
+  it("should set the value in redis with a ttl if given as a parsable number in a string", function(done) {
+    var target = new RedisCache({ maxAge: "1000" });
+    target.set("key", "value").then(function() {
+      target.client.cache.key.ttl.should.equal(1);
+      done();
+    });
+  });
+
+  it("should set the value in redis without a ttl if default options was given as an empty string", done => {
+    var target = new RedisCache({ maxAge: "" });
+    target.set("key", "value").then(function() {
+      assert(target.client.cache.key.ttl === null);
+      done();
+    });
+  });
+
+  ["unknown", "Infinity", "-Infinity", " ", {}, [], Infinity, Math.random].map(input => {
```

Comment:
Jag la in lite fler saker vi ska crasha vid uppstart på. Jag tror " " borde räknas in här eller hur?

Reply:
Ja!

Reply:
Snygg testkod också 🥇